webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -2216,22 +2178,25 @@ class SearchResponderV2 extends ResponderWithStandoffV2 {
             typeInspectionResult: GravsearchTypeInspectionResult <- gravsearchTypeInspectionRunner.inspectTypes(inputQuery.whereClause, requestingUser)
             whereClauseWithoutAnnotations: WhereClause = GravsearchTypeInspectionUtil.removeTypeAnnotations(inputQuery.whereClause)
 
-            // Preprocess the query to convert API IRIs to internal IRIs and to set inference per statement.
-
-            preprocessedQuery: ConstructQuery = QueryTraverser.transformConstructToConstruct(
```

Comment:
Did you get rid of this because in the old way, we just converted everything to internal assuming that we got it in simple schema. Now, we have to check if we get it in simple or complex.

Reply:
Yes.